test/avocado/testlib_avocado/libdisc.py

```diff
@@ -22,74 +22,75 @@
-        process.run("targetcli /iscsi/%s.%s:%s/tpg1/luns/ create /backstores/fileio/%s_%s" % (self.prefix, self.domain, name, self.domain, name), shell=True)
-        process.run("targetcli saveconfig", shell=True)
+        self.machine.execute("sudo targetcli /backstores/fileio/ create file_or_dev=/var/tmp/%s_%s.target size=10G sparse=true name=%s_%s" % (self.domain, name, self.domain, name))
+        self.machine.execute("sudo targetcli /iscsi/ create %s.%s:%s" % (self.prefix, self.domain, name))
+        self.machine.execute("sudo targetcli /iscsi/%s.%s:%s/tpg1/acls/ create %s" % (self.prefix, self.domain, name, self.initiatorname))
+        self.machine.execute("sudo targetcli /iscsi/%s.%s:%s/tpg1/luns/ create /backstores/fileio/%s_%s" % (self.prefix, self.domain, name, self.domain, name))
+        self.machine.execute("sudo targetcli saveconfig")
         self.targetlist.append(name)
 
     def createparttable(self, name, parttable='msdos'):
-        process.run("parted /var/tmp/%s_%s.target mktable %s" % (self.domain, name, parttable), shell=True)
+        self.machine.execute("sudo parted /var/tmp/%s_%s.target mktable %s" % (self.domain, name, parttable))
 
     def adddisc(self, targetsuffix):
         self.addtarget(targetsuffix)
-        process.run("iscsiadm -m discovery -t sendtargets -p %s" % self.ip, shell=True)
-        process.run("iscsiadm -m node --targetname=%s.%s:%s --login" % (self.prefix, self.domain, targetsuffix), shell=True)
-        tmp = process.run("iscsiadm -m node", shell=True)
-        print(tmp.stdout.decode("utf-8"))
-        tmp = process.run("sleep 5; iscsiadm -m session -P 3 | tail -1", shell=True)
-        tmp1 = re.search(r'Attached scsi disk\s+([a-z]*)\s+', tmp.stdout.decode("utf-8"))
+        self.machine.execute("sudo iscsiadm -m discovery -t sendtargets -p %s" % self.ip)
+        self.machine.execute("sudo iscsiadm -m node --targetname=%s.%s:%s --login" % (self.prefix, self.domain, targetsuffix))
+        tmp = self.machine.execute("sudo iscsiadm -m node")
+        print(tmp("utf-8"))
+        tmp = self.machine.execute("sleep 5; sudo iscsiadm -m session -P 3 | tail -1")
+        tmp1 = re.search(r'Attached scsi disk\s+([a-z]*)\s+', tmp("utf-8"))
         return "/dev/%s" % str(tmp1.group(1))
 
     def deldisc(self, name):
         targetname = "%s.%s:%s" % (self.prefix, self.domain, name)
-        process.run("iscsiadm -m node -T %s --portal %s -u" % (targetname, self.ip), shell=True)
+        self.machine.execute("sudo iscsiadm -m node -T %s --portal %s -u" % (targetname, self.ip))
         self.targetlist.remove(name)
 
     def clear(self):
         for foo in self.targetlist:
             self.deldisc(foo)
-        process.run("iscsiadm -m discovery -p %s -o delete" % self.ip, shell=True)
-        process.run("targetcli clearconfig confirm=True", shell=True)
-        process.run("targetcli saveconfig", shell=True)
+        self.machine.execute("sudo iscsiadm -m discovery -p %s -o delete" % self.ip)
```

Comment:
Is sudo needed here?

Reply:
for some commands probably not, but I'm lazy to ensure which command of them is necessary to run under root and which one could run under non privileged user. :-)